sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/AsyncResetTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics;
+
+import static io.opentelemetry.sdk.testing.assertj.metrics.MetricAssertions.assertThat;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.ObservableLongMeasurement;
+import io.opentelemetry.sdk.metrics.testing.InMemoryMetricReader;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+import org.junit.jupiter.api.Test;
+
+class AsyncResetTest {
+
+  @Test
+  void demonstrateAsyncResetBug() {
```

Comment:
This test duplicates the `TemporalMetricStorage#asynchronous_DeltaResetsAccumulations` test but in a clearer end-to-end fashion. I don't think we should keep this around, but thought it was helpful in illustrating the issue. 

I have an async observer that reports a different value for the attribute key `key` on each collection. The first collection, one metric data point is reported for `key=value1`. But on the second, when only `key=value1` is recorded metric data points are collected for both `key=value1` and `key=value2`.